### DISCUSSION TITLE
How to remove --rm from Quadlets?

### DISCUSSION THREAD

**Discussion by p5** [containers/podman#24806] [General]
I am needing to support a deployment of a third-party containerised application, and they require us to keep the same container across reboots.  If we were to try mounting the data inside volumes, we are not eligible for their support.  I know 🤦 ...

I can see the Quadlet is generated and includes the `--rm` flag, which is causing the container's filesystem to be wiped out on reboots.

How can we remove this `--rm` flag?

```
                CreateCommand": [
                    "/usr/bin/podman",
                    "run",
                    "--name=conjur_leader",
                    "--cidfile=/run/conjur-leader.cid",
                    "--replace",
                    "--rm",
                    "--cgroups=split",
                    "--sdnotify=conmon",
                    "-d",
                    "--security-opt",
                    "seccomp=/opt/cyberark/conjur/security/seccomp.json",
                    "--cap-add=audit_write",
                    "-v",
                    "/opt/cyberark/conjur/config:/etc/conjur/config:Z",
                    "-v",
                    "/opt/cyberark/conjur/security:/opt/cyberark/conjur/security:Z",
                    "-v",
                    "/opt/cyberark/conjur/backups:/opt/conjur/backup:Z",
                    "-v",
                    "/opt/cyberark/conjur/seeds:/opt/cyberark/conjur/seeds:Z",
                    "-v",
                    "/opt/cyberark/conjur/logs:/var/log/conjur:Z",
                    "--publish",
                    "443:443",
                    "--publish",
                    "444:444",
                    "--publish",
                    "5432:5432",
                    "--publish",
                    "1999:1999",
                    "conjur-appliance:13.4.0"
               ],
```

**Comment by rhatdan** [comment]
Why do they need the "Same" container across reboots?  Are they writing data to the container image instead of volumes?

**Reply by p5** [reply]
Yes...  That's exactly the reason.  
We've had the exact same discussion about it.  They seem to be doing everything their own way - shoving syslog, postgres, nginx and the app in a single container, refusing to provide any information about where the postgres data is stored, and outright saying things like "if you put the Postgres data in a volume, we will not be able to provide support".

Normally, I wouldn't mind and just add volumes myself, though we're paying $100k p/y for this software license and support alone, and need all the support we can get.

Their current recommendation is to use the systemd generate command, but this is obviously deprecated and we'd prefer to use Quadlets if possible.
https://docs.cyberark.com/conjur-enterprise/12.9/en/content/deployment/dap/dap-before-deploy-vm.htm?tocpath=Setup%7CSet%20up%20Conjur%20Enterprise%20(Docker%252FPodman)%7CPrepare%20Conjur%20environment%20for%20Docker%20or%20Podman%20deployment%7C_____1#:~:text=%24%20podman%20generate%20systemd%20%3Ccontainer%2Dname%3E%20%2D%2Dname%20%2D%2Dcontainer%2Dprefix%3D%22%22%20%2D%2Dseparator%3D%22%22%20%3E%20/etc/systemd/system/conjur.service

**Comment by Luap99** [comment]
We do not really support that with quadlet today and the persistence part has been discussed before: https://github.com/containers/podman/discussions/19620

**Reply by p5** [reply]
Thanks!  I'll close this and follow the linked discussion
